src/agent/coverage/src/code.rs

```diff
@@ -302,83 +301,113 @@ impl Region for Symbol {
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(transparent)]
-pub struct SymbolFilterSpec {
-    filters: HashMap<String, Filter>,
+pub struct CmdFilterDef {
+    defs: Vec<ModuleRuleDef>,
 }
 
-#[derive(Clone, Debug)]
-pub struct SymbolFilter {
-    /// Pre-compiled regex set for fast matching.
-    regexes: RegexSet,
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ModuleRuleDef {
+    pub module: String,
 
-    /// Maps module names (via regexes) to filters for the symbol names provided
-    /// by the matched module(s).
+    #[serde(flatten)]
+    pub rule: RuleDef,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+enum RuleDef {
+    Include { include: bool },
+    Exclude { exclude: bool },
+    Filter(Box<Filter>),
```

Comment:
This is boxed per a clippy lint: the `Filter` struct is on the order of 500 bytes, which forces any value of type `RuleDef` to be that size... event a variant that is otherwise just a `bool`!